packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -105,12 +105,12 @@ export function createBufferedTransformStream(): TransformStream<
   const flushBuffer = (controller: TransformStreamDefaultController) => {
     if (!pendingFlush) {
       pendingFlush = new Promise((resolve) => {
-        setTimeout(() => {
+        (async () => {
```

Comment:
Missing a semicolon in IIFE `;(async () => // ...`

I'm assuming that the use of `setTimeout` is intentional so as to create a buffer on the next event loop?

Reply:
The async function is properly encapsulated inside another function, so there's no risk of breakage if some tool inlines this file with other files. And five years ago StackOverflow had this:
>I answered this originally in 2013 and I should point out now that most modern javascript web tools are smart enough to handle this now, and with ES2015 and newer, there's better encapsulation, leading to less IIFEs. I haven't had to use an IIFE in years. – 
[Erty Seidohl](https://stackoverflow.com/users/374601/erty-seidohl)
 [Apr 17, 2018 at 20:03](https://stackoverflow.com/questions/17978883/what-is-the-purpose-of-a-semicolon-before-an-iife#comment86789191_17978926)
 
The buffer requirement is satisfied with the first let statement. flushBuffer is just trying to do its job of flushing the buffer, but setTimeout is the worst way to do this, as the callback is put on the bottom of the stack. SetImmediate would be better, as it goes on the top of the stack, but the anonymous async function is clearly the winner, as the tests show.